chain/chain/src/chain.rs

```diff
@@ -2689,6 +2696,9 @@ impl Chain {
                 ));
             }
             root_proofs.push(root_proofs_cur);
+            (target_shard_id, _) = self
```

Comment:
Because `target_shard_id` was already known for `block_hash`, and this function moves shard id 1 block back, so it must happen in the end. (Anyway I reverted changes in the file)